contracts/SimpleQuests.sol

```diff
@@ -408,29 +433,90 @@ contract SimpleQuests is Initializable, AccessControlUpgradeable {
+        questIndexes[questID] = questTemplates[tier].length - 1;
+        if (supply > 0) {
+            require(deadline > 0, "Missing deadline");
+            questSupplies[questID] = supply;
+            questDeadlines[questID] = deadline;
+        }
+    }
+
+    // TODO: Not sure how to manage created rewards and setting the week. Maybe it should be a single function that takes in info and timestamp?
+    function addReward(ItemType rewardType, uint256 rewardRarity, uint256 rewardAmount, address rewardExternalAddress, uint256 reputationAmount) public restricted {
+        uint256 rewardID = nextRewardID++;
+        rewards[rewardID] = Reward(rewardID, rewardType, rewardRarity, rewardAmount, rewardExternalAddress, reputationAmount);
+        emit RewardAdded(rewardID);
     }
 
-    function addNewPromoQuestTemplate(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
-        RewardType rewardType, Rarity rewardRarity, uint256 rewardAmount, uint256 reputationAmount) public restricted {
-        uint256 questID = addNewQuest(tier, requirementType, requirementRarity, requirementAmount, rewardType, rewardRarity, rewardAmount, reputationAmount);
-        questTemplates[uint8(tier) + 10].push(questID);
+    function setWeeklyReward(uint256 id, uint256 timestamp) public restricted {
+        require(timestamp > 0, "Missing timestamp");
+        uint256 week = timestamp / 1 weeks;
+        weeklyRewards[week] = id;
+        emit WeeklyRewardSet(id, week);
     }
 
-    function addNewQuest(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
-        RewardType rewardType, Rarity rewardRarity, uint256 rewardAmount, uint256 reputationAmount) internal returns (uint256 questID) {
-        questID = nextQuestID++;
-        quests[questID] = Quest(questID, tier,
-            requirementType, requirementRarity, requirementAmount,
-            rewardType, rewardRarity, rewardAmount, reputationAmount);
+    // TODO: What should be the identifier here?
+    function generateRewardWeeklySeed(uint256 rewardID) assertQuestsEnabled public {
+        safeRandoms.requestSingleSeed(address(this), RandomUtil.combineSeeds(SEED_REWARD_WEEKLY, rewardID));
+    }
+
+    function claimWeeklyReward() public returns (uint256[] memory weeklyRewardIDs) {
+        uint256 currentCompletions = weeklyCompletions[msg.sender][now / 1 weeks];
+        require(weeklyRewardClaimed[msg.sender][now / 1 weeks] == false, "Reward already claimed");
+        require(currentCompletions >= vars[VAR_WEEKLY_COMPLETIONS_GOAL], "Not enough weekly completions");
+        uint256 rewardID = weeklyRewards[now / 1 weeks];
+        weeklyRewardIDs = rewardWeekly(rewardID);
+        emit WeeklyRewardClaimed(msg.sender, rewardID, weeklyRewardIDs);
+        weeklyRewardClaimed[msg.sender][now / 1 weeks] = true;
+    }
+
+    function rewardWeekly(uint256 rewardID) private returns (uint256[] memory) {
```

Comment:
I duplicated it because of the EXPERIENCE type 
![image](https://user-images.githubusercontent.com/34208222/155758199-9581e833-0910-483a-8bb5-d890e1771d46.png)
Dunno how to omit the parameter in a nice way here

Reply:
hmm. The char id param likely has to stay. But maybe we can pass 0 when it's a weekly reward claim and never have xp weekly rewards (we didnt plan to anyway).
ALTERNATIVELY we can make the player pick which character is claiming the reward, in which case we have a legit character ID passed

Reply:
The problem with this is also with the map we're getting the struct from, in case of quest it's Quest, and weekly it's Reward
![image](https://user-images.githubusercontent.com/34208222/155763976-322ef038-ccdf-4b34-bd6f-360d04ecbb4c.png)